packages/core-utils/src/base-service.ts

```diff
@@ -38,6 +37,12 @@ export class BaseService<T> {
     }
 
     this.logger.info('Service is initializing...')
+
+    if (typeof window === 'undefined') {
```

Comment:
i wonder if base-service is the right place to do this conditional loading, since there's almost no case that this file will be used by the browser, and the error comes from building the frontends after having installed contracts

do you have a frontends branch with core-utils installed that shows this error? this comment https://github.com/siimon/prom-client/issues/191#issuecomment-417281496 makes it sound like we should be doing the optional imports / builds there instead, is that what you thought too @snario

Reply:
I think we can enable this library to be used in browser or node without requiring the consumer to have to think about it.

Reply:
I agree that `base-service` isn't the best place for this and am not sure how this solves the problem exactly. If I understand correctly, `prom-client` will break if its imported into the frontend code? Can we use a webpack rule to block the file from being imported?

Reply:
maybe we could not auto export `metrics` and `base-service` from index? and instead have an`./internal` path?